### PR TITLE
release: sign Linux tarball and .NET tool

### PR DESCRIPTION
**Pull request by ldennington**
Update release workflow to:

1. Split the Linux `build.sh` script and the .NET tool `pack-tool.sh` script to more closely align with our compilation and packaging for macOS.
2. Sign Linux tarball in addition to the Debian package.and sign/validate/publish .NET tool `.nupkg`.
3. Sign/validate/publish the .NET tool `.nupkg` with every release.

I validated these changes with [this successful workflow run](https://github.com/ldennington/git-credential-manager/actions/runs/3651681219) in my fork.

### CODE REVIEW

**Review by mjcheetham — APPROVED** [review]
Looks good! One typo spotted.

File: src/linux/Packaging.Linux/build.sh [hunk 1]
```diff
@@ -41,208 +36,47 @@ case "$i" in
...
-    dpkg-deb --build "$DEBROOT" "$DEBPKG" || exit 1
+    echo "Install complete."
+else
+    # Layout and pack
+    "$INSTALLER_SRC/layout.sh" --configuration="$CONFIGURATION" --output="$PAYLOAD" || exit 1
+    "$INSTALLER_SRC/pack.sh" --configuration="$CONFIGURATION" --payload="$PAYLOAD" --version="$VERSION" || exit 1
 fi
 
-echo $MESSAGE
+echo "Build of Packagining.Linux complete."
```
[the remark this answers is not in the document]
Comment (ldennington) [reply src/linux/Packaging.Linux/build.sh:82]: I have literally had so many problems with this. My fingers simply refuse to type the word `Packaging` correctly!

**Comment by z279** [comment]
https://github.com/GitCredentialManager/git-credential-manager/releases/tag/v2.0.886

**Comment by z279** [comment]
https://github.com/GitCredentialManager/git-credential-manager/releases/tag/v2.0.886

**Comment by z279** [comment]
https://github.com/GitCredentialManager/git-credential-manager/releases/tag/v2.0.886

**Review by z279 — COMMENTED** [review]
https://github.com/GitCredentialManager/git-credential-manager/releases/tag/v2.0.886

**Review by emmanuelle2089 — COMMENTED** [review]
my personally ogf inforn

**Review by emmanuelle2089 — COMMENTED** [review]
only  samey

**Pull request merged**